packages/jasmine/src/coverage_runner.js

```diff
@@ -0,0 +1,99 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const fs = require('fs');
+const path = require('path');
+const crypto = require('crypto');
+const childProcess = require('child_process');
+
+/**
+ * This is designed to collect the coverage of one target, since in nodejs
+ * and using NODE_V8_COVERAGE it may produce more than one coverage file, however bazel expects
+ * there to be only one lcov file. So this collects up the v8 coverage json's merges them and
+ * converts them to lcov for bazel to pick up later.
+ * TODO: The functionality in this script could be put into _lcov_merger then it would apply to
+ * any tool reporting coverage not just jasmine
+ */
+async function main() {
+  const result = childProcess.spawnSync(
+      process.execPath,
+      [...process.execArgv, __dirname + '/jasmine_runner.js', ...process.argv.slice(2)],
+      {stdio: 'inherit'});
+
+  const exitCode = result.status;
+
+  // 0 indicates success
+  // so if it's anything tthen exit here since bazel only collects coverage on success
+  if (exitCode !== 0) {
+    process.exit(exitCode);
+  }
+
+  if (!coverageEnabled) {
+    process.exit(exitCode);
+  }
+
+  const coverageDir = process.env.COVERAGE_DIR;
+  const outputFile = process.env.COVERAGE_OUTPUT_FILE;
+  const sourceFileManifest = process.env.COVERAGE_MANIFEST;
+
+  const instrumentedSourceFiles = fs.readFileSync(sourceFileManifest).toString('utf8').split('\n');
+
+  // c8 will name the output report file lcov.info
+  // so we give it a dir that it can write to
+  // later on we'll move and rename it into output_file as bazel expects
+  const tmpdir = process.env['TEST_TMPDIR'];
+  const c8OutputDir = path.join(tmpdir, crypto.randomBytes(4).toString('hex'));
+  fs.mkdirSync(c8OutputDir);
+
+  const includes = instrumentedSourceFiles
+                       // we can only cover js files
+                       .filter(f => ['.js', '.jsx', '.ts', '.tsx'].includes(path.extname(f)))
+                       // the actual coverage written by v8 will only inclue urls to .js
+                       // so map inputs to their outputs
+                       .map(f => {
+                         const p = path.parse(f);
+                         return path.format({...p, base: null, ext: '.js'});
+                       });
+
+  // onyl require in c8 when we're actually going to do some coverage
+  const c8 = require('c8');
+  // see https://github.com/bcoe/c8/blob/master/lib/report.js
+  // for more info on this function
+  // TODO: enable the --all functionality
+  await new c8
+      .Report({
+        include: includes,
+        exclude: [],
+        reportsDirectory: c8OutputDir,
+        // tempDirectory as actually the dir that c8 will read from for the v8 json files
+        tempDirectory: coverageDir,
+        resolve: '',
+        // TODO: maybe add an attribute to allow more reporters
+        // or maybe an env var?
```

Comment:
Would more reporters be useful since we're only interested in the lcov data for bazel coverage?

Reply:
I've found it helpful to have either `text` it `text-summary` to instantly report on the unit testing suit you're working on so you can aim to cover more paths.
From what I understand going the bazel route will be pretty slow and wont work with ibazel